source/aws_profile.c

```diff
@@ -1541,14 +1541,17 @@ struct aws_string *aws_get_config_file_path(
 AWS_STATIC_STRING_FROM_LITERAL(s_default_profile_env_variable_name, "AWS_PROFILE");
 
 struct aws_string *aws_get_profile_name(struct aws_allocator *allocator, const struct aws_byte_cursor *override_name) {
-
+    /**
+     * The default profile must be used if the user does NOT specify a profile.
+     * the default profile can be overridden using the AWS_PROFILE environment variable.
+     */
```

Comment:
We should update the docs in header file to list the order the resolution.

Reply:
I think it would be more clear to list it something like
```
/**
* Profile name is resolved in the following order.
* 1. If the override_path variable is provided, use that.
* 2. Check `AWS_PROFILE` and use that if it is not empty.
* 3. Use the `default` profile name.
*/
```